src/antithesis_sdk/assertions.py

```diff
@@ -361,29 +372,123 @@ def assert_raw(
         assert_id,
     )
 
+def _readlines(fname: str, verbose=False) -> list[str]:
+    all_lines = []
+    with open(fname) as f:
+        for line in f:
+            if verbose:
+                print(line, end="")
+            all_lines.append(line)
+    return all_lines
+
+def _get_subdirs(dir_path: str) -> list[str]:
+    if not os.path.isdir(dir_path):
+        return []
+    walk_results = next(os.walk(dir_path))
+    return walk_results[1] # directories at index=1, files at index=2
+
+
+def _get_module_list(file_path: str) -> list[str]:
+    """Reads all lines in file_path, looking for
+    any comment lines that contain:
+    module_name = '<module_name>'
+    parse these lines and return a list of the
+    module names found,
+    """
+
+    listed_modules = []
+    rx = re.compile(r"^\s*#\s*module_name\s*=\s*(\S*)\s*")
+    lines = _readlines(file_path)
+    for line in lines:
+        maybe_match = rx.match(line)
+        if maybe_match is not None:
+            matched_repr = maybe_match.group(1) 
+            module_name = literal_eval(matched_repr)
+            listed_modules.append(module_name)
+    return listed_modules
+
+
+def get_grade(py_catalog_path: str, module_list: list[str]) -> float:
+    """Read and parse the python source catalog in `py_catalog_path`
+    to obtain a module_list.  Count the number of modules that can 
+    be loaded from this list, and return the overall grade of loadable 
+    modules found in the range 0.0 to 1.0
+    """
+    num_modules = float(len(module_list))
+    num_found = 0
+    with open(catalog_path, "r") as fp:
+        for module_name in module_list:
+            this_spec = find_spec(module_name)
+            my_module = module_from_spec(this_spec) 
+            if this_spec is not None:
+                num_found = num_found + 1
+    return num_found/num_modules
+
+def _get_instrumentation_folder(from_path: str) -> Optional[str]:
+    """Determines which subfolder of `from_path` contains the
```

Comment:
I think we could probably add a bit more detail to these doc strings as it might be difficult to understand what we did here in a couple months time!